src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IPatternSwitchCase.cs

```diff
@@ -0,0 +1,679 @@
+  Sections: ISwitchCase (1 case clauses, 1 statements) (OperationKind.SwitchCase) (Syntax: 'case T y:/* ... break;')
+        Clauses: IPatternCaseClause (Label Symbol: case T y:) (OperationKind.PatternCaseClause) (Syntax: 'case T y:')
+            Pattern: IDeclarationPattern (Declared Symbol: T y) (OperationKind.DeclarationPattern) (Syntax: 'T y')
+        Body: IBranchStatement (BranchKind.Break) (OperationKind.BranchStatement) (Syntax: 'break;')
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyOperationTreeAndDiagnosticsForTest<CasePatternSwitchLabelSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact, WorkItem(19927, "https://github.com/dotnet/roslyn/issues/19927")]
+        public void TestPatternCaseClause_DynamicTypePatternDeclaration()
+        {
+            string source = @"
+using System;
+class X
+{
+    void M<T>(object x) where T : class
+    {
+        switch (x)
+        {
+            /*<bind>*/case dynamic y:/*</bind>*/
+                break;
+        }
+    }
+}
+";
+            string expectedOperationTree = @"
+ISwitchStatement (1 cases) (OperationKind.SwitchStatement, IsInvalid) (Syntax: 'switch (x) ... }')
+  Switch expression: IParameterReferenceExpression: x (OperationKind.ParameterReferenceExpression, Type: System.Object) (Syntax: 'x')
+  Sections: ISwitchCase (1 case clauses, 1 statements) (OperationKind.SwitchCase, IsInvalid) (Syntax: 'case dynami ... break;')
+        Clauses: IPatternCaseClause (Label Symbol: case dynamic y:) (OperationKind.PatternCaseClause, IsInvalid) (Syntax: 'case dynamic y:')
+            Pattern: IDeclarationPattern (Declared Symbol: dynamic y) (OperationKind.DeclarationPattern, IsInvalid) (Syntax: 'dynamic y')
+        Body: IBranchStatement (BranchKind.Break) (OperationKind.BranchStatement) (Syntax: 'break;')
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // CS8208: It is not legal to use the type 'dynamic' in a pattern.
+                //             /*<bind>*/case dynamic y:/*</bind>*/
+                Diagnostic(ErrorCode.ERR_PatternDynamicType, "dynamic").WithLocation(9, 28)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<CasePatternSwitchLabelSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact, WorkItem(19927, "https://github.com/dotnet/roslyn/issues/19927")]
+        public void TestPatternCaseClause_MixedDeclarationPatternAndConstantPatternClauses()
+        {
+            string source = @"
+using System;
+class X
+{
+    void M(object x)
+    {
+        switch (x)
+        {
+            case null:
+                break;
+            /*<bind>*/case X y:/*</bind>*/
+                break;
+        }
+    }
+}
+";
+            string expectedOperationTree = @"
+ISwitchStatement (2 cases) (OperationKind.SwitchStatement) (Syntax: 'switch (x) ... }')
```

Comment:
Yes, we still want to validate the IOperation tree structure for the pattern case label when the switch statement also has a constant expression label. We will need to edit the baselines for almost all tests in this file though.